klt-cert-manager/controllers/keptnwebhookcontroller/keptnwebhookcertificate_controller.go

```diff
@@ -52,11 +44,57 @@ func (r *KeptnWebhookCertificateReconciler) Reconcile(ctx context.Context, reque
 
 	r.ctx = ctx
 
-	_, err := r.getMutatingWebhookConfiguration()
+	mutatingWebhookConfiguration, err := r.getMutatingWebhookConfiguration()
 	if err != nil {
 		r.Log.Error(err, "could not find mutating webhook configuration")
 	}
 
+	crds := &apiv1.CustomResourceDefinitionList{}
+	crds, err = r.getCRDConfigurations()
+	if err != nil {
+		r.Log.Error(err, "could not find CRDs")
+	}
+
+	certSecret := newCertificateSecret()
+
+	err = certSecret.setSecretFromReader(r.ctx, r.ApiReader, namespace, r.Log)
```

Comment:
why do we need the `ApiReader` in addition to the `Client` that we also have in the reconciler?